src/main/java/fi/aalto/cs/apluscourses/intellij/actions/CourseProjectAction.java

```diff
@@ -130,25 +152,33 @@ public void actionPerformed(@NotNull AnActionEvent e) {
       return;
     }
 
-    String language = courseProjectViewModel.languageProperty.get();
+    String language = Objects.requireNonNull(courseProjectViewModel.languageProperty.get());
     if (!tryCreateCourseFile(project, selectedCourseUrl, language)) {
       return;
     }
 
-    startAutoInstallsWithRestart(course, !courseProjectViewModel.userOptsOutOfSettings(), project);
+    Awaitable installDone = startAutoInstallsWithRestart(course, project);
+    Objects.requireNonNull(installDone); // because of checkstyle stupidity
 
     if (!tryImportProjectSettings(project, course)) {
       return;
     }
 
-    if (!courseProjectViewModel.userOptsOutOfSettings()) {
-      tryImportIdeSettings(course);
+    if (!courseProjectViewModel.userOptsOutOfSettings() && !tryImportIdeSettings(project, course)) {
```

Comment:
"remove this explession that always evaluates to `false`" says sonar for `!tryImportIdeSettings(project, course)`

Reply:
Sonar is wrong. It doesn't understand `try-catch`es.